cluster-autoscaler/core/scaledown/pdb/pdb.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pdb
+
+import (
+	"fmt"
+
+	policyv1 "k8s.io/api/policy/v1"
+)
+
+// PdbRemainingDisruptions stores how many discuptiption is left for pdb.
+type PdbRemainingDisruptions struct {
+	remainingDisruptions map[string]int
+	pdbForName           map[string]policyv1.PodDisruptionBudget
+}
+
+// GetDisruptions return how many disruptions is left for pdb.
+func (p PdbRemainingDisruptions) GetDisruptions(pdb policyv1.PodDisruptionBudget) (int, error) {
+	pdbName := pdb.GetObjectMeta().GetNamespace() + pdb.GetObjectMeta().GetName()
+	pdbDisruptions, ok := p.remainingDisruptions[pdbName]
+	if !ok {
+		return 0, fmt.Errorf("There is no pdb %s in list", pdbName)
+	}
+	return pdbDisruptions, nil
+}
+
+// UpdatePdb updates the remaining disrubtions for pdb.
+func (p PdbRemainingDisruptions) UpdatePdb(pdb policyv1.PodDisruptionBudget, disrubtions int) error {
+	pdbName := pdb.GetObjectMeta().GetNamespace() + pdb.GetObjectMeta().GetName()
```

Comment:
We should have some kind of separator here, otherwise pdb `foo` in namespace `bar` is the same as pdb `oo` in namespace `barf`.